src/query/catalog/src/table.rs

```diff
@@ -71,6 +71,15 @@ pub trait Table: Sync + Send {
 
     fn get_table_info(&self) -> &TableInfo;
 
+    fn check_can_modify_table(&self) -> Result<()> {
```

Comment:
We can move this check to a privilege check?
https://github.com/datafuselabs/databend/pull/7721

The prepared work is you should bind the share_name to table info in the binder.